Stage 2

irmo12 opened the pull request.

irmo12 committed:
font-faces

irmo12 committed:
init header

irmo12 committed:
import app fix

irmo12 committed:
header css WIP

irmo12 committed:
init routes, footer

irmo12 committed:
fix svgs imports

irmo12 committed:
init about, searchbar

irmo12 committed:
footer init, preloader init

irmo12 committed:
style searchform

irmo12 committed:
style WIP

irmo12 committed:
stlyes WIP

irmo12 committed:
style preloader searching, not yet in harmony with 'no results' state

irmo12 committed:
done preloader

irmo12 committed:
style footer, search

irmo12 committed:
init newscard, newscardlist

irmo12 committed:
popup init

irmo12 committed:
popups WIP

irmo12 committed:
add saved news components, popup fixes, routing changed, placeholders for articles

irmo12 committed:
styling articles, saved and list

irmo12 committed:
style cards, fixing saved route styles and header

irmo12 committed:
removed redundant newslist in savednews route

irmo12 committed:
remove some comments, consider removing infopopup component

irmo12 committed:
infopopup component init

irmo12 committed:
search form responsive

irmo12 committed:
lift up small screens, useContext instead of state and effect

irmo12 committed:
fix routing, styles

irmo12 committed:
routing, vis bugs

irmo12 committed:
Rename preloader.css to Preloader.css

FiLLa-FaS commented:
Hi! Thanks for submitting your project for the first review - it looks pretty good already and I like your work with jsx components. But we still can improve the project, so I left several comments below, please fix them to get more points! Good luck!✨

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout and get out of the grid in the layout.
    * Unfortunately the project doesn't match the figma page on all screen resolutions in many aspects, it has too many inconsistencies, so I can't mark them all - you need to check all sizes, spaces, colors and text properties for all elements in the project by yourself (please check all pages and popups). You can use pixelPerfect plugin to make it easier https://www.welldonecode.com/perfectpixel/
    * All news card images are broken on both pages
- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correctly on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
    * The page scrolls horizontally on 320px - 750px screen resolutions (`overflow: hidden` is not a solution of scroll, you need to remove this styles and fix the scrolling changing the width and spaces of elements)
- [ ] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
    * `sign up` button inside sign in popup doesn't work, so I can't get the access to the sign up popup
    * All footer links to external resources should be opened in a new tab
- [ ] <!--4.2--> Classes are named according to BEM specifications.
    * `popup__form-field_error`, `..._logged-in` should be key-value modifiers instead of boolean https://en.bem.info/methodology/quick-start/#key-value
    * `popup__form-field-group_invisible`, `preloader_hidden` should be used only with its base classes
    * Please don't use capital letters in class names
- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g. a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * You need to use correct semantic tags for `main`, `footer`, `navigation-container`, `preloader`, `news-card-list`, `about`, `footer-signature`, `saved-news` blocks
    * `search-form__search-bar` should be a form
- [x] <!--4.2--> The correct approach is chosen for positioning elements, and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [ ] <!--4.2--> The popup windows have opening and closing functionality implemented.
    * I can't open the popup from the page instead of changing the state manually in the component. Popup should be opened by clicking the `sign in` button in the header, and `sign up` should be opened by clicking the switch button inside the sign in popup
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [ ] <!--4.2--> There are no linting errors or warnings.
    * You should fix eslint warnings

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * All needed states are specified in the mockup, you need to implement all states for all links and buttons according to the `state` iframe of the figma page. Please note, that all buttons and links should have pointer cursor
    * News card on the main page should have booking button, and card on `saved-news` page should have trash button - each button should have itw own hover and marked state according to the figma design
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * Not all `font-family` styles have alternative fonts
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 41

irmo12 committed:
minor nav fixes WIP, smalscreen conttext change to 768

irmo12 committed:
wip phone

irmo12 committed:
popup WIP

irmo12 committed:
popup fixes WIP

irmo12 committed:
popup fixes

irmo12 committed:
WIP gallery fixes

irmo12 committed:
various visual fixes

irmo12 committed:
review fixes various, main needs restructuring

FiLLa-FaS commented:
Hi! Thanks for the corrections, your project looks much better. Hovewer there're still some issues that need to be fixed to get more points. Good luck!✨

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
    * All links in header and footer should have white or black text color according to the mockup
    * Many text elements have incorrect font-weight (they might have correct `font-weight` style but fonts are linked incorrectly)
    * Incorrect width of `search-form__search-bar` on 1440 viewport
    * Incorrect text color of `news-card__source`
    * Incorrect background color of the footer section
    * Incorrect text properties of all elements inside header and width of sign button on 768 viewport
    * Please check width of all items inside `search-form` and gaps between elements on 768 viewport
    * Incorrect height of the news cards on 768 viewport
    * `about` section looks broken on 768 viewport - incorrect side spaces, text properties, sizes and spaces of all items inside the section on 768 viewport
    * Incorrect side spaces inside footer section on 768 viewport
    * The main page is broken on 320 viewport - you need to check all sizes, spaces and text properties of all items inside the page for this screen resolutions
    * Header should have only logo and navigation menu on mobile screen resolutions https://pasteboard.co/6TOsogyxwp91.png
    * `saved-news` page has many mismatches with the design on all breakpoints screen resolutions - you need to check all layouts, sizes, spaces and text properties of all elements inside the page
    * Popup looks incorrectly on 320 viewport (screenshot of how it should look on 320 viewport https://pasteboard.co/l98qZVuykgfq.png)
- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
    * Side spaces of the page on more than 1440px should have white background https://pasteboard.co/FcSWtkZEZJFE.png
    * The main page scrolls horizontally on 320px - 418px screen resolutions
    * `saved-news` page scrolls horizontally on all screen resolutions
    * Popup and close button shouldn't extend outside the screen on intermediate screen resolutions
- [ ] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
    * `sign in` popup should be opened when I clicked `sign in` button (currently sign up popup opens)
- [ ] <!--4.2--> Classes are named according to BEM specifications.
    * `popup__form-field-group_invisible`, `preloader_hidden` should be used only with its base classes
- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * You need to use correct semantic tags for `main`, `footer`, `navigation-container`, `preloader`, `news-card-list`, `about`, `footer-signature`, `saved-news` blocks
    * `search-form__search-bar` should be a form
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [ ] <!--4.2--> There are no linting errors or warnings.
    * You should fix all eslint warnings

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * All needed states are specified in the mockup, you need to implement all states for all links and buttons according to the `state` iframe of the figma page. Please note, that all buttons and links should have pointer cursor
    * News card on the main page should have booking button, and card on `saved-news` page should have trash button - each button should have itw own hover and marked state according to the figma design
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * Not all `font-family` styles have alternative fonts
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 46

irmo12 committed:
review fixes

irmo12 committed:
header visual fixes WIP

irmo12 committed:
markup various

irmo12 committed:
saved-grid fixes

irmo12 committed:
footer fixes

irmo12 committed:
vis fixes grid

irmo12 committed:
header fixes, init phoneHeaderExtension

irmo12 committed:
vis fixes, sbout WIP

irmo12 committed:
about&header fixes WIP

irmo12 committed:
about+footer fixes

irmo12 committed:
phoneheader problems

irmo12 committed:
header refactor, vis fixes

irmo12 committed:
header phone, next refactor button for DNR

irmo12 committed:
refactor headerbutton as comp to achieve DNR

irmo12 committed:
infopopup WIP

irmo12 committed:
refactor w/o redundant 'savednews' comp

irmo12 committed:
header nav-link bottom border fix, phone fixes WIP

irmo12 committed:
semantics. also for the record, its DRY not DNR. DNR stands for do not resuscitate lol

irmo12 committed:
PR fixes done?

irmo12 committed:
phone header shenanigans

irmo12 committed:
node malfunctions, had to reinstall header styles problems?

irmo12 committed:
routing mishap fixed, header fixes

irmo12 committed:
vis fixes and rev fixes complete. finally implemented full phone header functionality: user can log in or out and switch pages. however svg icon color problem persists

irmo12 committed:
init BE-FE interface

irmo12 committed:
user can sign up and log in. header on small screen misbehaving

irmo12 committed:
header fixes

irmo12 committed:
header logic fixes

irmo12 committed:
infopopup z-index

irmo12 committed:
review mode

irmo12 committed:
dev init

irmo12 committed:
get articles init

irmo12 committed:
saveorDel init

FiLLa-FaS commented:
Hi! Your project has been improved a lot, all the critical issues can be found below, keep up the good work!✨

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
    * `header_logged_in` looks incorrectly (header should have background depending on its route - main page should have background image and `saved-articles` page should have white background)
    * Footer has incorrect elements order
    * Mismatches with the mockup for main page on 1440 viewport https://pasteboard.co/nk2z0gPnpUjb.png https://pasteboard.co/X6MXX2w9EAhi.png https://pasteboard.co/yO7PpNzOUCWh.png
    * Mismatches with the mockup for main page on 768 viewport https://pasteboard.co/6pfCyndJC4YH.png https://pasteboard.co/j7qjnN5VnxwJ.png https://pasteboard.co/WSMezUyazc6t.png 
    * Mismatches with the mockup for main page on320 viewport https://pasteboard.co/OxSsoxvfHrF1.png https://pasteboard.co/QIK2LQa0dxZS.png
    * Mismatches with the mockup for `saved-news` page on 1440 viewport https://pasteboard.co/WoeL57PVKJ8a.png, 768 viewport https://pasteboard.co/GeiJ0FL9wNZi.png, 320 viewport https://disk.yandex.ru/i/jnYqqbR6GLkKEA
    * Mobile nav menu looks incorrectly on 320 viewport (please check the height of the `header_show-nav`, sizes of the elements and gaps between items)
    * Incorrect size of the news card and its content on 768 and 320 viewports - you should check all sizes, spaces and text properties
    * All links in header and footer should have white or black text color according to the mockup
    * Popup looks incorrectly on 320 viewport (screenshot of how it should look on 320 viewport https://pasteboard.co/l98qZVuykgfq.png)
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
    * header tabs shou;dn't extend outside the screen on intermediate screen resolutions (please check 550px - 580px screen resolutions)
    * the `saved-news` page scrolls horizontally on 320px - 584px screen resolutions
    * Side spaces of the page on more than 1440px should have white background https://pasteboard.co/FcSWtkZEZJFE.png
    * The page scrolls horizontally on 320px - 550px screen resolutions when the mobile nav menu is open
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * `list` tag doesn't exist, please use correct semantic tag
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [ ] <!--4.2--> There are no linting errors or warnings.
    * You should fix all eslint warnings

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * `news-card__save-del` button should have different icon depending on route - home page should have booking icon, saved-news page should have trash icon, hover and active state for booking button can be found in the figma IU kit
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * Not all `font-family` styles have alternative fonts or generic font-families
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 62

irmo12 committed:
header colors fixed

irmo12 committed:
rev fixes ongoing

irmo12 committed:
rev fixes, fonts bugged

irmo12 committed:
vis fixes

irmo12 committed:
merging

irmo12 committed:
fonts fixed, overrode font-face rules by accident, update correct image for main, header overflows and sizing fixes

irmo12 committed:
various vis fixes

irmo12 committed:
vis fixes

irmo12 committed:
cards 1440 WIP

irmo12 committed:
changed news card to grid setup

irmo12 committed:
rev fixes

irmo12 committed:
grid breaking card needs fixing

irmo12 committed:
must solve grid text overflow

irmo12 committed:
getting there 320

irmo12 committed:
almost there

irmo12 committed:
crossed fingers, news-card buttons later

FiLLa-FaS commented:
Hi! You've made a lot of changes, great job🙌. Several small issues are left, a few more fixes and the project will be accepted. Good luck!

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
    * Background image inside `main__header-search-container` section doesn't correspond the mockup
    * All links in header and footer should have white or black text color according to the mockup https://pasteboard.co/F0l0XGCO8WjA.png
    * Mismatches with the mockup for saved-news page on 768 viewport https://pasteboard.co/s7KDlZvc7aEU.png, 320 viewport https://pasteboard.co/NaoXf3JHE7VO.png
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * `list` tag doesn't exist, please use correct semantic tag
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.
    * You should fix all eslint warnings

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * `news-card__save-del` button should have different icon depending on route - home page should have booking icon, saved-news page should have trash icon, hover and active state for booking button can be found in the figma IU kit
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 77

irmo12 committed:
ENOENT error?

irmo12 committed:
semantic HTML fix, eslint warnings

irmo12 committed:
removed def styles from li elements,
match mockup img (bg,opacity),
correct footer links def styles when visited,
match mockup in 320, 768 for saved news.

FiLLa-FaS commented:
Hi! You've got more than 80 points, so I'm glad to accept your project. You still can fix the rest issues later. Good luck on the next stage 🙌🚀

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
    * All links in `header` and `footer` should have white or black text color according to the mockup https://pasteboard.co/F0l0XGCO8WjA.png
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
    * `news-card__save-del` button should have different icon depending on route - home page should have booking icon, saved-news page should have trash icon, hover and active state for booking button can be found in the figma IU kit
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 85
